CompilerSource/parsing/ast.h

```diff
@@ -209,6 +276,104 @@ class AST {
     WithStatement(PNode object_, PNode body_): object{std::move(object_)}, body{std::move(body_)} {}
   };
 
+  enum class DeclaratorType {
+    ABSTRACT, NON_ABSTRACT, MAYBE_ABSTRACT
```

Comment:
In a function declaration like `int foo(int, int x)` you don't know if the parameters have names before parsing their declarators, so this flag allows the parser to parse a name only if it exists. Its not really necessary, I just like the strictness of the parse it implies because it would be just as easy to have it always be `MAYBE_ABSTRACT` and then error in the following passes depending on the contents of `Declarator::name`